web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -178,11 +182,21 @@ public void init(FilterConfig config) {
         }
     }
 
+    private void configureWebAppNameContextInitializer(String appName, TelemetryConfiguration configuration) {
+        for (ContextInitializer ci : configuration.getContextInitializers()) {
+            if (ci instanceof WebAppNameContextInitializer) {
+                ((WebAppNameContextInitializer)ci).setAppName(appName);
```

Comment:
Please correct me if I am thinking write. 

ContextInitializers -> TelemetryInitializers -> Processors (This is the pipeline correct) in this case it should be fine. If this is not the case and ContextInitializers fall after TelemetryInitializers than this will screw up the springboot app name (set by TelemetryInitializer) which is not the same as the servlet context parsed name and should be set actually in case of springboot instead of servlet context parsed name. Double check and confirm this.

Reply:
Yup make sense to update it to context initializer. Now the question is, will boot initializer override the one in the core sdk, then it should be perfectly fine. We can try and double check that it is the case !

Reply:
That's a good question, but I don't think they'll conflict. The `WebAppNameContextInitializer` doesn't need to be added to a spring-boot app, but I think it's possible for the `WebAppNameContextInitializer` to replace the spring-boot one.

As it stands now, with no changes, you have to explicitly add the WebAppNameContextInitializer so it should function properly. Let's have a discussion about the best approach here when you're back.

Reply:
I think this would work fine. I double checked the code in the starter. Essentially we create a barebone TelemetryConfiguration() object there and inject all the initializers and modules etc using DI. 

https://github.com/Microsoft/ApplicationInsights-Java/blob/cf042c35fdae23479e84765962580e307a8aadef/azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsTelemetryAutoConfiguration.java#L114

Reply:
As discussed, the flow is a bit different and the spring boot initializer overwrites the app name from the WebAppNameContextInitializer, if both are enabled. The latter initializer is required in config to work, so it should be rare to see both initializer enabled.